web/src/main/webapp/my-app/layout/controllers.js

```diff
@@ -125,327 +152,131 @@ define(['angular', 'jquery'], function(angular, $) {
             // save new layout order
             saveLayoutOrder(nextIndex, $scope.layout.length, widget.nodeId);
             // log change
-            $scope.logEvent('keyboardMove',
+            $scope.logMoveEvent('keyboardMove',
               widget.fname,
               nextIndex,
               currentIndex);
           }
         }
       };
 
-      /**
-       * After a widget is moved, save the new layout using
-       * the given information
-       * @param dropIndex {Number} Where the widget ended up
-       * @param length {Number} Length of the layout array
-       * @param nodeId {String} ID of the moved widget
-       */
-      var saveLayoutOrder = function(dropIndex, length, nodeId) {
-        // identify previous and next widgets
-        var previousNodeId =
-          dropIndex !== 0 ? $scope.layout[dropIndex - 1].nodeId : '';
-        var nextNodeId =
-          dropIndex !== length - 1 ? $scope.layout[dropIndex + 1].nodeId : '';
-        // call layout service to save
-        layoutService.moveStuff(dropIndex,
-          length, nodeId, previousNodeId, nextNodeId);
-      };
-
-      /**
-       * Find an array object with the given attribute/value pair
-       * @param array {Array} The array to iterate on
-       * @param attribute {String} The name of the attribute
-       * @param value {String} The value to match on
-       * @return {number} Index of the desired item or -1
-       */
-      var findLayoutIndex = function(array, attribute, value) {
-        for (var i = 0; i < array.length; i+= 1) {
-          if (array[i][attribute] === value) {
-            return i;
+      // Listen for removal event
+      $scope.$on('REMOVE_WIDGET',
+        /**
+         * Listen for widget removal event, then show confirmation toast
+         * if one is not already showing.
+         * @param event {Object} The angularjs event object
+         * @param data {Object} Data about the widget being removed
+         */
+        function(event, data) {
+          // Make sure no toasts are already open
+          if (!$scope.isToastDisplayed) {
```

Comment:
Is silently ignoring `REMOVE_WIDGET` event the right behavior in the case where that event fires with a toast already displayed? Or if that were to happen is the right thing to go ahead and stack up another toast? (Looks like such an event should never happen.)